### ISSUE TITLE
Exception, possibly while saving to the OSF (refresh_node referenced before assignment)

### ISSUE THREAD

**Issue by smathot** [dschreij/opensesame-extension-osf#12]
I received this message as part of an "Oops, I did it again". I had just opened an experiment, and then saved it (also to the OSF). Shortly after I received the error, but I'm not 100% sure that the two are linked.

~~~ python
Traceback (most recent call last):
  File "/home/sebastiaan/git/QOpenScienceFramework/QOpenScienceFramework/manager.py", line 395, in <lambda>
    callback, *args, **kwargs
  File "/home/sebastiaan/git/QOpenScienceFramework/QOpenScienceFramework/manager.py", line 854, in __reply_finished
    callback(reply, *args, **kwargs)
  File "/home/sebastiaan/git/opensesame-extension-osf/opensesame_extensions/OpenScienceFramework/OpenScienceFramework.py", line 1325, in __prepare_experiment_sync
    selectedTreeItem=refresh_node,
UnboundLocalError: local variable 'refresh_node' referenced before assignment
~~~

**Comment by dschreij** [comment]
Thanks. I'll look into this soon. I guess this is what they mean with "callback hell" in the sense that it's hard to establish a chronological order in creating and referencing variables 😉

**Comment by dschreij** [comment]
Did you refresh (part of) the tree while this process was ongoing by any chance?

**Comment by smathot** [comment]
> Did you refresh (part of) the tree while this process was ongoing by any chance?

No, for sure not. I had switched to another application, and when I switched back the error was already there. That's why I'm not sure about the what, when, and how.

**Comment by dschreij** [comment]
Should be fixed with 3a08aaa57daed6c8482f76df74d574ec176a0ee2
Quite sloppy, I checked for the (very rare) occasion that a refresh node wasn't supplied with a try...except... as I should, but I didn't handle things well in the except section. This uninstantiated variable error should no longer happen.
